src/libraries/Common/src/Interop/Windows/WebSocket/Interop.Structs.cs

```diff
@@ -46,38 +46,36 @@ internal struct CloseBuffer
         internal struct HttpHeader
         {
             internal string Name;
-            internal uint NameLength;
             internal string Value;
-            internal uint ValueLength;
 
             [CustomMarshaller(typeof(HttpHeader), MarshalMode.ManagedToUnmanagedIn, typeof(Marshaller))]
             [CustomMarshaller(typeof(HttpHeader), MarshalMode.ElementIn, typeof(Marshaller))]
             public static class Marshaller
             {
-                public static Native ConvertToUnmanaged(HttpHeader managed)
+                public static WEB_SOCKET_HTTP_HEADER ConvertToUnmanaged(HttpHeader managed)
                 {
-                    Native n;
+                    WEB_SOCKET_HTTP_HEADER n;
                     n.Name = Marshal.StringToCoTaskMemAnsi(managed.Name);
-                    n.NameLength = managed.NameLength;
+                    n.NameLength = (uint)managed.Name.Length;
                     n.Value = Marshal.StringToCoTaskMemAnsi(managed.Value);
-                    n.ValueLength = managed.ValueLength;
+                    n.ValueLength = (uint)managed.Value.Length;
```

Comment:
Pre-existing your change, but is it correct that the managed string length will always be the same as the Ansi output length?

Reply:
This code seems to assume that all strings are ASCII. There is a lot of additional cleanup that can be done on this (e.g. the strings do not actually need to be converted between Unicode and ASCII at all). I have limited my change to just deleting unnecessary code around the interop that I have run into.